File: UniversityApiBackend/DTOs/Students/StudentDetailDTO.cs
using UniversityApiBackend.DTOs.Courses;

namespace UniversityApiBackend.DTOs.Students
{
    public class StudentDetailDTO
    {
        public int Id { get; set; }
      
        public DateTime Dob { get; set; }
        public UserDTO User { get; set; } = new UserDTO();
        public AddressDTO Address { get; set; } = new AddressDTO();

        public List<CourseListDTO> Courses { get; set; }
    }
}

File: Frontend/front-university/src/app/services/api/models/studentDetailDTO.ts
import { AddressDto } from "./address-dto";
import { Course } from "./course";
import { CourseListDTO } from "./courseListDTO";
import { User } from "./user";

export interface StudentDetailDTO {
  id?:      number;
  dob?:     Date;
  user?:    User;
  address?:  AddressDto;
  courses?: CourseListDTO[];
}


